src/contracts/pods/EigenPod.sol

```diff
@@ -169,133 +135,144 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
-     * @param validatorFields are the fields of the "Validator Container", refer to consensus specs
-     * @dev For more details on the Beacon Chain spec, see: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
+     * @dev Create a checkpoint used to prove this pod's active validator set. Checkpoints are completed 
+     * by submitting one checkpoint proof per ACTIVE validator. During the checkpoint process, the total 
+     * change in ACTIVE validator balance is tracked, and any validators with 0 balance are marked `WITHDRAWN`.
+     * @dev Once finalized, the pod owner is awarded shares corresponding to:
+     * - the total change in their ACTIVE validator balances
+     * - any ETH in the pod not already awarded shares.
+     * @dev A checkpoint cannot be created if the pod already has an outstanding checkpoint. If
+     * this is the case, the pod owner MUST complete the existing checkpoint before starting a new one.
      */
-    function verifyBalanceUpdates(
-        uint64 oracleTimestamp,
-        uint40[] calldata validatorIndices,
-        BeaconChainProofs.StateRootProof calldata stateRootProof,
-        bytes[] calldata validatorFieldsProofs,
-        bytes32[][] calldata validatorFields
-    ) external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_BALANCE_UPDATE) {
-        require(
-            (validatorIndices.length == validatorFieldsProofs.length) && (validatorFieldsProofs.length == validatorFields.length),
-            "EigenPod.verifyBalanceUpdates: validatorIndices and proofs must be same length"
-        );
-
-        // Balance updates should not be "stale" (older than VERIFY_BALANCE_UPDATE_WINDOW_SECONDS)
+    function startCheckpoint() 
+        onlyEigenPodOwner() 
+        onlyWhenNotPaused(TODO) 
+        afterRestaking(uint64(block.timestamp)) /// TODO - this is the wrong condition
+    {
         require(
-            oracleTimestamp + VERIFY_BALANCE_UPDATE_WINDOW_SECONDS >= block.timestamp,
-            "EigenPod.verifyBalanceUpdates: specified timestamp is too far in past"
+            currentCheckpointTimestamp == 0, 
+            "EigenPod.startCheckpoint: must finish previous checkpoint before starting another"
         );
 
-        // Verify passed-in beaconStateRoot against oracle-provided block root:
-        BeaconChainProofs.verifyStateRootAgainstLatestBlockRoot({
-            latestBlockRoot: eigenPodManager.getBlockRootAtTimestamp(oracleTimestamp),
-            beaconStateRoot: stateRootProof.beaconStateRoot,
-            stateRootProof: stateRootProof.proof
+        // Snapshot pod balance at the start of the checkpoint. Once the checkpoint is finalized,
+        // this amount will be added to the total validator balance delta and credited as shares.
+        uint256 podBalanceWei = 
+            address(this).balance
+                - nonBeaconChainETHBalanceWei
```

Comment:
should this just be autorestaked?

Reply:
`nonBeaconChainETHBalanceWei`? Yeah I'm all for it.

Also worth noting that the pod balance calculated here truncates any wei amount so that it sits in storage as gwei. we should figure out if that's okay or not! (the amount isn't fully lost; you can always top up and prove another checkpoint)